include/networkit/io/NetworkitBinaryWriter.hpp

```diff
@@ -30,13 +30,19 @@ class NetworkitBinaryWriter final : public GraphWriter {
 
 public:
     NetworkitBinaryWriter(uint64_t chunks = 32,
-                          NetworkitBinaryWeights weightsType = NetworkitBinaryWeights::autoDetect);
+                          NetworkitBinaryWeights weightsType = NetworkitBinaryWeights::autoDetect,
+                          bool preserveEdgeIndex = false);
```

Comment:
Why is this `false` by default? I think we should default to true if the graph has edge indices.

What happens if the users sets `true` here but the graph has no edge indices?